src/WP_CLI/CommandWithMeta.php

```diff
@@ -31,40 +31,68 @@
 	 * [--format=<format>]
 	 * : Accepted values: table, csv, json, count. Default: table
 	 *
+	 * [--orderby=<fields>]
+	 * : Set orderby which field.
+	 * ---
+	 * default: meta_key
+	 * options:
+	 *  - meta_key
+	 *  - meta_value
+	 * ---
+	 *
+	 * [--order=<order>]
+	 * : Set ascending or descending order.
+	 * ---
+	 * default: asc
+	 * options:
+	 *  - asc
+	 *  - desc
+	 * ---
 	 * @subcommand list
 	 */
 	public function list_( $args, $assoc_args ) {
+		global $wpdb;
 
 		list( $object_id ) = $args;
 
-		$keys = ! empty( $assoc_args['keys'] ) ? explode( ',', $assoc_args['keys'] ) : array();
+		// Selected keys.
+		$keys          = ! empty( $assoc_args['keys'] ) ? explode( ',', $assoc_args['keys'] ) : array();
+		$object_id     = $this->check_object_id( $object_id );
 
-		$object_id = $this->check_object_id( $object_id );
+		// Default sorting parameter.
+		$orderby_field = 'meta_key';
+		$sort          = Utils\get_flag_value( $assoc_args, 'order' );
+
+		// Get metadata.
+		if ( 'user' === $this->meta_type ) {
+			// Where clause.
+			$where    = $wpdb->prepare( " WHERE `$wpdb->usermeta`.`user_id` = %d ", $object_id );
+			$metadata = $wpdb->get_results( "SELECT `user_id`,`meta_key`,`meta_value` FROM `$wpdb->usermeta` {$where} ORDER BY `$wpdb->usermeta`.`$orderby_field` $sort" );
+		} else {
+			// Where clause.
+			$where    = $wpdb->prepare( " WHERE `$wpdb->postmeta`.`post_id` = %d ", $object_id );
+			$metadata = $wpdb->get_results( "SELECT `post_id`,`meta_key`,`meta_value` FROM `$wpdb->postmeta` {$where} ORDER BY `$wpdb->postmeta`.`$orderby_field` $sort" );
+		}
```

Comment:
Per my original comment, can we perform the orderby with PHP instead of SQL, so we can continue to use `get_metadata()`?